libs/js-sdk/src/utils/fetchFeaturesConfiguration.ts

```diff
@@ -28,6 +35,38 @@ export async function fetchFeaturesConfigurationViaHttp(
         },
     })
 
+    if (response.status === 429) {
+        // Too many requests
+        const retryAfterHeader = response.headers.get('Retry-After')
+        const retryAfterInt = retryAfterHeader
+            ? parseInt(retryAfterHeader, 10)
+            : 60
+        const retryAfter =
+            retryAfterHeader && !retryAfterInt
+                ? new Date(retryAfterHeader)
+                : new Date()
+
+        if (retryAfterInt) {
+            const retryAfterTime = retryAfter.getTime() + retryAfterInt * 1000
+            retryAfter.setTime(retryAfterTime)
+        }
+
+        return {
+            etag,
+            retryAfter,
+            error: new TooManyRequestsError(
```

Comment:
If we throw here, the etag could go on the error itself?

I'm not a huge fan of returning errors in properties, unless it's a pattern used throughout the SDK.

Not super strong on changing back though, just putting the thoughts down. Happy to leave it

Reply:
I came to same conclusion myself, I have changed it.